src/pcp/atop/netproc.c

```diff
@@ -0,0 +1,79 @@
+/*
+** Copyright (C) 2020 Red Hat.
+**
+** This program is free software; you can redistribute it and/or modify it
+** under the terms of the GNU General Public License as published by the
+** Free Software Foundation; either version 2, or (at your option) any
+** later version.
+**
+** This program is distributed in the hope that it will be useful, but
+** WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+** See the GNU General Public License for more details.
+*/
+
+#include <pcp/pmapi.h>
+#include "atop.h"
+#include "photoproc.h"
+#include "netprocmetrics.h"
+
+void
+netatop_probe(void)
+{
+	pmID	pmids[TASK_NET_NMETRICS];
+
+	int sts = pmLookupName(TASK_NET_NMETRICS, netprocmetrics, pmids);
+	if (sts == TASK_NET_NMETRICS)
+		supportflags |= NETATOP;
+	else
+		supportflags &= ~NETATOP;
+}
+
+static void
+netproc_update_task(struct tstat *task, int pid, pmResult *rp, pmDesc *dp)
+{
+	task->net.tcpsnd = extract_count_t_inst(rp, dp, TASK_NET_TCPSND, pid);
+	task->net.tcprcv = extract_count_t_inst(rp, dp, TASK_NET_TCPRCV, pid);
+	task->net.tcpssz = extract_count_t_inst(rp, dp, TASK_NET_TCPSSZ, pid);
+	task->net.tcprsz = extract_count_t_inst(rp, dp, TASK_NET_TCPRSZ, pid);
+
+	task->net.udpsnd = extract_count_t_inst(rp, dp, TASK_NET_UDPSND, pid);
+	task->net.udprcv = extract_count_t_inst(rp, dp, TASK_NET_UDPRCV, pid);
+	task->net.udpssz = extract_count_t_inst(rp, dp, TASK_NET_UDPSSZ, pid);
+	task->net.udprsz = extract_count_t_inst(rp, dp, TASK_NET_UDPRSZ, pid);
+}
+
+void
+netproc_update_tasks(struct tstat **tasks, unsigned long taskslen)
+{
+	static int 	setup;
+	static pmID	pmids[TASK_NET_NMETRICS];
+	static pmDesc	descs[TASK_NET_NMETRICS];
+	pmResult	*result;
+	char		**insts;
+	int		*pids;
+	int		netproc_insts_len, i, j;
+
+	if(!setup) {
+		setup_metrics(netprocmetrics, pmids, descs, TASK_NET_NMETRICS);
+		setup = 1;
+	}
+
+	fetch_metrics("task_netproc", TASK_NET_NMETRICS, pmids, &result);
+	netproc_insts_len = get_instances("task_netproc", TASK_NET_TCPSND, descs, &pids, &insts);
+
+	for (i=0; i < netproc_insts_len; i++)
+	{
+		if (pmDebugOptions.appl0)
+			fprintf(stderr, "%s: updating net info of process %d: %s\n",
+				pmGetProgname(), pids[i], insts[i]);
+
+		// for each instance of the netproc metrics, we need to find the index inside the tasks array
+		// TODO: better algorithm to avoid O(m*n) complexity?
+		for(j=0; j < taskslen; j++) {
```

Comment:
Coding style (braces, whitespace).

Reply:
I've seen https://github.com/performancecopilot/ansible-pcp/blob/master/.travis/runblack.sh in the Ansible roles repo the other day. Black [1] is a code formatter with almost no settings, which takes any unformatted code and replaces it with nicely formatted code. No matter which editor you use and how non-uniform you place whitespaces etc., you will always end up with exactly the same formatting (and tries to produce the smallest diffs possible). afaics it takes the AST and formats it, i.e. the tool ignores all previous formatting. The same thing is also done (and enforced) in every Grafana plugin (using a tool called prettier [2]).

Is there interest to enforce (and automatically apply) a consistent styleguide to all our C code as well? A similar tool I could find for C code is clang-format [3].

[1] https://github.com/psf/black
[2] https://prettier.io/
[3] https://clang.llvm.org/docs/ClangFormat.html

Reply:
| Is there interest to enforce (and automatically apply) a consistent styleguide to all our C code as well? 

Its come up from time to time, and generally the answer is "no, let sleeping dogs lie"  :)

Reply:
hmm, I'd be open for some modernization of the codebase ;) (also looking at the makefiles - there are much more sophisticated tools available in 2020) - might be worth a discussion on the mailing list after the next release(s)